docs/CaliptraHardwareSpecification.md

```diff
@@ -2521,6 +2522,116 @@ Data vault is a set of generic scratch pad registers with specific lock function
+**CALIPTRA_OCP_LOCK_PROGRESS_REG**
+- KeyVault slots 0-15: Standard use-cases
+- KeyVault slots 16-23: Standard use-cases
+- Write-1-to-lock
+- ROM must set this after performing LOCK derivations (HEK, EPK, etc)
+- CALITPRA_OCP_LOCK_EN + CALIPTRA_MODE_SUBSYSTEM macro is the write-enable for this bit
+- Prevents interactions between slots (STANDARD slots) and (LOCK slots)
+- Enables Key Release operation
+
+### 5.3 Add AES Write Path
+Decrypted Keys (or “unwrapped”/ “ready” keys, e.g. MPK) may be written to KV slots.  
+**MEK** must be written to KV slot 23. Enforced in HW by detecting AES-ECB decrypt operation with KvSlots16 (MDK) as key.
+
+### 5.7 Add KeyVault access rules and filtering logic
+- ROM is responsible for choosing dest slot for HEK seed
+- INSTEAD, Slot 23 PREVENTED as destination for all DOE operations
+  - Slot 23 (place for MEK) restricted so ONLY AES can write to it (when OCP LOCK in progress)
+  - Slot 22 also used to store full derived HEK, which overwrites HEK seed (ROM requirement)
+  - Locked for writes until warm reset (ROM requirement, must lock after each reset)
+  - Locked for copying until warm reset (ROM requirement, must lock after each reset)
+- There must not be a way from KV0-16 to KV10-15 or from KV10-15 to KV9-0 if OCP parameter and OCP_LOCK_PROGRESS_REG, which is set only by ROM, are high.
+- IF (OCP LOCK mode enabled) THEN (KV23 shall never be used as input to other crypto operations, only as source for Key Release). Prevents known values propagating to other KV slots.
+- IF (OCP LOCK mode enabled) AND (AES-ECB decrypt operation) AND (key = KV16) THEN (dest = KV23)  
+  ELSE (dest=FW) << Preventing AES writes to other KV slots is important to avoid malicious firmware getting known values into KV.
+
+**KeyVault gains intelligence:**
+- Track which crypto made most recent read request.
+- On Write, compute if the write destination is legal for the read.
+- RULE: No cryptos with KV access operate in parallel. KV won’t track this, but caliptra_top enforces via HW_ERROR
+
+### HEK seed fuse register
+Used as input to HEK seed deobfuscation operation
+
+### 7 HEK Seed Deobfuscation
+Hardware path from Ratchet Fuse Register (Obf HEK Seed) as input to DOE (with OBF_KEY), with output stored to KV slot 9.
+- Slot 9 is locked for write on DOE operation.
+- Slot 9 is locked for use immediately after being used to derive HEK (which is placed into a slot 10-14).
+
+### 8 Input Straps for MEK address and size
+Writable until FUSE_WR_DONE, then locked, same as other straps.  
+Input addr strap is full address to MEK – FW can derive SFR_BASE from this strap as needed.
+Input size strap indicates byte-count (dword-count preferred by HW) of MEK to program to MEK address.
+
+### 9 Key Release
+Hardware path allowing KV slot 15 to be written to SoC using DMA block.
+- Only permitted when CALIPTRA_OCP_LOCK_EN (synth macro) is set and if CALIPTRA_OCP_LOCK_PROGRESS_REG (sticky w1set bit set by Caliptra ROM) are set.
```

Comment:
Can probably say only permitted when OCP_LOCK_IN_PROGRESS set since this bit can't be set unless CALIPTRA_OCP_LOCK_EN is set. 

Also CALIPTRA_OCP_LOCK_EN is not a synth macro it is a strap.